nucypher_cli/demos/finnegans-wake-demo.py

```diff
@@ -3,55 +3,40 @@
 
 # WIP w/ hendrix@3.0.0
 
-import binascii
 import datetime
-import logging
 import sys
 
 import maya
+from umbral.keys import UmbralPublicKey
 
+from nucypher.blockchain.eth.chains import Blockchain
 from nucypher.characters import Alice, Bob, Ursula
 from nucypher.data_sources import DataSource
-# This is already running in another process.
-from nucypher.network.middleware import RestMiddleware
-from umbral.keys import UmbralPublicKey
 
-root = logging.getLogger()
-root.setLevel(logging.DEBUG)
 
-ch = logging.StreamHandler(sys.stdout)
-ch.setLevel(logging.INFO)
-formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
-ch.setFormatter(formatter)
-root.addHandler(ch)
+##############################################
+# This is already running in another process.
+##############################################
 
-teacher_dht_port = sys.argv[2]
-teacher_rest_port = int(teacher_dht_port) + 100
-with open("examples-runtime-cruft/node-metadata-{}".format(teacher_rest_port), "r") as f:
-    f.seek(0)
-    teacher_bytes = binascii.unhexlify(f.read())
-URSULA = Ursula.from_bytes(teacher_bytes, federated_only=True)
-print("Will learn from {}".format(URSULA))
+BLOCKCHAIN = Blockchain.connect()
 
-# network_middleware = SandboxRestMiddleware([URSULA])
+URSULA = Ursula.from_config()
 
 #########
 # Alice #
 #########
 
-ALICE = Alice(network_middleware=RestMiddleware(),
-              known_nodes=(URSULA,),  # in lieu of seed nodes
-              federated_only=True,
-              always_be_learning=True)  # TODO: 289
+ALICE = Alice.from_config()
 
 # Here are our Policy details.
-policy_end_datetime = maya.now() + datetime.timedelta(days=5)
+policy_end_datetime = maya.now() + datetime.timedelta(days=201)
```

Comment:
eh?  201?

Reply:
This is a debugging step for the development of the decentralized Finnegan's wake demo - `NotEnoughUrsulas` is raised because there are not enough Ursulas for the selected duration, even though the population is large enough.